modules/plugin/shapefile/src/main/java/org/geotools/data/shapefile/ShapefileFeatureSource.java

```diff
@@ -382,12 +370,8 @@ protected int getCountInternal(Query query) throws IOException {
             reader.setFilter(filter);
         }
 
-        // setup the target bbox if any, and the generalization hints if available
+        // setup the generalization hints if available
         if (q != null) {
-            if (bbox != null && !bbox.isNull()) {
```

Comment:
Hi @aaime, I know your advices are right, but I do not know other mode to avoid that the BBOX defined in a part of mixed alphanumeric-spatial filter rejects features that do not share these envelope but matches the alphanumeric clause. I managed to explain my intentions?